### ISSUE TITLE
[Mainet] Peer connected on snap without compatible support 

### ISSUE THREAD

**Issue by supamongkonR** [bnb-chain/bsc#2235]
when I run node from information in document and I following the step that recommend to run geth from snapshot not from genesis, 
and I've got bsc.log like this  

i use this command to run my node

```
./geth --config ./config.toml --datadir ./node  --cache 8000 --rpc.allow-unprotected-txs --history.transactions 0 &
```
and this is result
```
t=2024-02-22T06:59:05+0000 lvl=info msg="Imported new chain segment"             number=35,363,272 hash=0xa3b687e76cb17367ac6e2d336a6415e7ecf9b777da59137723a8d604cf33f5d3 miner=0x685B1ded8013785d6623CC18D214320b6Bb64759 blocks=63 txs=5555  mgas=635.869  elapsed=8.144s    mgasps=78.074  age=1mo4d9h   triediffs="134.67 MiB" triedirty="32.88 MiB"  trieimutabledirty="0.00 B"
t=2024-02-22T06:59:08+0000 lvl=warn msg="Snapshot extension registration failed" peer=a38b6811 err="peer connected on snap without compatible eth support"
t=2024-02-22T06:59:11+0000 lvl=warn msg="Snapshot extension registration failed" peer=7f4e0609 err="peer connected on snap without compatible eth support"
t=2024-02-22T06:59:11+0000 lvl=warn msg="Snapshot extension registration failed" peer=7f41201d err="peer connected on snap without compatible eth support"
t=2024-02-22T06:59:11+0000 lvl=warn msg="Snapshot extension registration failed" peer=7f72d5a6 err="peer connected on snap without compatible eth support"
t=2024-02-22T06:59:13+0000 lvl=info msg="Imported new chain segment"             number=35,363,339 hash=0x2c01c195a1f65365283e71d0d46aba864d1a75f3d98734b996bdffe7ab4da7de miner=0x8b6C8fd93d6F4CeA42Bbb345DBc6F0DFdb5bEc73 blocks=67 txs=5638  mgas=640.970  elapsed=8.018s    mgasps=79.934  age=1mo4d9h   triediffs="131.45 MiB" triedirty="51.73 MiB"  trieimutabledirty="0.00 B"
t=2024-02-22T06:59:14+0000 lvl=warn msg="Snapshot extension registration failed" peer=4a3fd4a2 err="peer connected on snap without compatible eth support"

```
and then it's seem like this issue is solve by run 
```
geth --datadir node init genesis.json
```
from https://github.com/bnb-chain/bsc/issues/776
and they explain, if we start by don't init and then chain state default will use ETH mainet that why it not compatible 

so my question is why we need to syncing from genesis when we got snapshot and how to solve it 

or I need to run geth init first then start node with sanpshot ?

**Comment by weiihann** [comment]
The snapshot that is mentioned in the documentation refers to the copy of an existing node, which is a bunch of folders and files to run a node.

The snapshot that is mentioned in the log that you've shown refers to the [snap protocol](https://github.com/ethereum/devp2p/blob/master/caps/snap.md). 

`Snapshot extension registration failed` is a normal log, but I agree we should do something to optimize the protocol connections. IF you don't do init genesis.json, it will by default use the BSC mainnet chain ID, the existing issue you referred to is outdated. Based on your logs, your node seems to be importing blocks as usual.

In other words, don't worry too much, your node is good to go :)

**Comment by supamongkonR** [comment]
Thanks you so much 

and sometimes I got this
```
t=2024-02-22T05:56:31+0000 lvl=eror msg="Bsc extension Handshake failed"         peer=dacbfb38 err="read timeout"
```

**Comment by weiihann** [comment]
> Thanks you so much
> 
> and sometimes I got this
> 
> ```
> t=2024-02-22T05:56:31+0000 lvl=eror msg="Bsc extension Handshake failed"         peer=dacbfb38 err="read timeout"
> ```

It's also normal. The log means that the BSC subprotocol (used to propagate votes) failed when negotiating the connection with a peer. Not going to affect node syncing.

**Comment by Ava00c** [comment]
> And when become validator, I need to stop node right ?and then after stake start with this command by documentation
> 
> ```
> geth --config ./config.toml --datadir ./node --syncmode snap -unlock {accounts to sign txs, including your mining account at least} --miner.etherbase {the address of your mining account} --password password.txt --blspassword blspassword.txt --mine --vote --allow-insecure-unlock --cache 18000
> ```

Yes, but BSC is POSA consensus, if you want to be a mainnet validator, you can ref https://www.bnbchain.org/en/staking for more information.

**Comment by weiihann** [comment]
I think you don't need to stop your node after staking, provided that the command used is the same.

**Comment by supamongkonR** [comment]
You mean I can use staking command and then done,  while node is running right ?

**Comment by weiihann** [comment]
> You mean I can use staking command and then done, while node is running right ?

Yup

**Comment by supamongkonR** [comment]
Thanks you so much.
